### ISSUE TITLE
Question: Long Running Hooks - Yes or No?

### ISSUE THREAD

**Issue by Inspyro** [helm/helm#10582]
Hi I have a question regarding the usage of **hooks**, as it is not very clear from the documentation:
Is it recommended to design hooks that run for a **very long** period of time (e.g. _hours up to days_?) or should those rather be designed as **standard-jobs**?

**Comment by mattfarina** [comment]
Chart hooks are [designed for specific tasks which are reflected in their name](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks). Is there a need for a `post-install` hook that runs for hours or days? A hook is part of the process for Helm (a package manager) to install, upgrade, or delete an instance of a chart. Helm will wait on hooks to complete as part of this process. So, a hook that runs for hours will leave Helm running (with a long timeout) as it processes a hook for installing, upgrading, or deleting.

Most of the time, a long running hook (for hours or days) isn't a good idea. For example, if the network drops between the Helm client and the Kubernetes API the release information (stored by default as a secret) won't be updated to show the process completed.

If something is part of an app it shouldn't be a hook. That should be a job or other Kubernetes resource since it's not part of the install, upgrade, or delete process.

Does that help?

**Comment by Inspyro** [comment]
Thanks that answers the question. 

Essentially I'm talking about stuff like **db-data-migration** that can take a long time.
I guess one could still say the data-migration is part of the upgrade process, but I assume according to your answer this would rather be considered a part of the application
